code-monkey/src/main.py

```diff
@@ -11,35 +12,35 @@
 console = Console()
 
 @instrument("main")
-def main(debug: bool = False) -> None:
+async def main(debug: bool = False) -> None:
     setup_logging(debug)
     print("[bold green]Welcome to the AI Agent Selector![/bold green]")
-    
+
     agent_choices = [
         ("Coder", Coder),
         ("CodeAnalyst", CodeAnalyst),
         ("Manager", Manager)
     ]
-    
+
     menu_items = [f"{i+1}. {name}" for i, (name, _) in enumerate(agent_choices)]
     terminal_menu = TerminalMenu(menu_items, title="Choose an agent to run:")
     menu_entry_index = terminal_menu.show()
-    
+
     if menu_entry_index is None:
         print("[bold red]No selection made. Exiting...[/bold red]")
         return
-    
+
     agent_name, agent_class = agent_choices[menu_entry_index]
```

Comment:
This still needs a refactor to use the new `run_agent_main` function, but that's on me!